### PR TITLE
Feature/T9-update-project-dependencies

### PR DESCRIPTION
**Pull request by MarcoHuebner**
Adresses #9 as well as #6 
- Updated the GitHub workflow with current poetry version
- Updated all packages to current major version - pandas should not have introduced any breaking changes, since most or even all of the processing is done in dicts and the pandas [options](https://pandas.pydata.org/docs/user_guide/options.html) stayed the same
- Update poetry lock file

### CODE REVIEW

**Comment by pmayd** [comment]
Did you investigated why the cicd fails for flake8?

**Comment by MarcoHuebner** [comment]
> Did you investigated why the cicd fails for flake8?

Not sure what exactly failed the plugin installations (docstrings installed but missing), but flake8 6.1.0 seemed to be the reason. After reverting to 3.9.2, it is running smoothly again

File: .github/workflows/run-tests.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
     - name: Run poetry image
       uses: abatilo/actions-poetry@v2.0.0
       with:
-          poetry-version: 1.1.13
+          poetry-version: 1.6.1
```
Comment (pmayd) [inline .github/workflows/run-tests.yaml:36]: can we reintroduce the matrix option here like you did in the big PR about sphinx? much cleaner

**Review by pmayd — APPROVED** [review]
LGTM

**Comment by pmayd** [comment]
I changed the target to dev as PR are never merged into main directly

File: .github/workflows/run-tests.yaml (hunk 1 above)
Comment (MarcoHuebner) [reply .github/workflows/run-tests.yaml:36]: Is included in the latest dev-branch hotfix, as I had to fix the workflow anyway

**Pull request merged**